addons/l10n_ro_edi_stock/models/stock_picking.py

```diff
@@ -0,0 +1,118 @@
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class Picking(models.Model):
+    _inherit = 'stock.picking'
+
+    l10n_ro_edi_stock_interface_id = fields.Many2one(
+        comodel_name='l10n_ro.edi.stock.etransport.interface',
+        delegate=True,
+        ondelete='cascade',
+        default=None,
+    )
+
+    l10n_ro_edi_stock_enable = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable')
+    l10n_ro_edi_stock_enable_send = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_send')
+    l10n_ro_edi_stock_enable_fetch = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_fetch')
+    l10n_ro_edi_stock_enable_send_info = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_send_info')
+    l10n_ro_edi_stock_enable_amend = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_amend')
+
+    l10n_ro_edi_stock_fields_readonly = fields.Boolean(compute='_compute_l10n_ro_edi_stock_fields_readonly')
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+
+        for picking in records.filtered(lambda record: record.company_id.country_id.code == 'RO'):
+            if not picking.l10n_ro_edi_stock_interface_id:
+                picking.l10n_ro_edi_stock_interface_id = picking.env['l10n_ro.edi.stock.etransport.interface'].create({'picking_id': picking.id})
+            elif not picking.l10n_ro_edi_stock_interface_id.picking_id or picking.l10n_ro_edi_stock_interface_id.picking_id.id != picking.id:
+                picking.l10n_ro_edi_stock_interface_id.picking_id = picking
+
+        return records
+
+    @api.returns(None, lambda value: value[0])
+    def copy_data(self, default=None):
+        self.ensure_one()
+        if default is None:
+            default = {}
+
+        if self.l10n_ro_edi_stock_interface_id:
+            copy_data = self.l10n_ro_edi_stock_interface_id.copy_data()[0]
+            interface_copy = self.env['l10n_ro.edi.stock.etransport.interface'].create(copy_data)
+            default['l10n_ro_edi_stock_interface_id'] = interface_copy.id
+
+        return super().copy_data(default)
+
+    @api.onchange('l10n_ro_edi_stock_operation_type')
+    def _reset_location_type(self):
+        self.l10n_ro_edi_stock_interface_id._reset_location_type()
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_start_loc_type')
+    def _reset_start_location_data(self):
+        self.l10n_ro_edi_stock_interface_id._reset_location_data('start')
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_end_loc_type')
+    def _reset_end_location_data(self):
+        self.l10n_ro_edi_stock_interface_id._reset_location_data('end')
```

Comment:
It is necessary for the following case: when the start or end location type is set to 'location' and the operation_type is changed. The `reset_location_type` on the interface will then set the start and end location types to 'location'. Now if the value of the start or end location types where already set to 'location' then it will not trigger the corresponding onchange function. 

This means that we need to specify the `l10n_ro_edi_stock_operation_type` in all 3 of the onchange functions